starter_code/src/movies.js

```diff
@@ -1,19 +1,98 @@
 /* eslint no-restricted-globals: 'off' */
 
-// Iteration 1: All rates average - Get the average of all rates with 2 decimals 
+// Iteration 1: All rates average - Get the average of all rates with 2 decimals
+
+function ratesAverage(arr) {
+  const ratingsSum = arr.reduce(function(a, b) {
+    return a + (parseFloat(b.rate) || 0); // Or !! => true = 1 and false = 0
+  }, 0);
+  return Math.round((ratingsSum / arr.length) * 100) / 100;
+}
 
- 
 // Iteration 2: Drama movies - Get the average of Drama Movies
 
+function dramaMoviesRate(arr) {
+  const dramaMovies = arr.filter(movie => movie.genre.includes("Drama"));
+  if (!dramaMovies.length) {
+    return 0;
+  }
+  return ratesAverage(dramaMovies);
+}
 
 // Iteration 3: Ordering by duration - Order by time duration, ascending (in growing order)
 
+function orderByYear(arr) {
+  return arr.sort(function(a, b) {
+    if (a.year !== b.year) {
+      return a.year - b.year;
+    }
+    return a.title.localeCompare(b.title);
+  });
+}
 
 // Iteration 4: Steven Spielberg. The best? - How many movies did STEVEN SPIELBERG direct
 
+function howManyMovies(arr) {
+  const dramaSpilbergMovies = arr.filter(movie => {
+    if (
+      movie.genre.includes("Drama") &&
+      movie.director.includes("Steven Spielberg")
+    ) {
+      return true;
+    }
+  });
+  if (!dramaSpilbergMovies.length) {
+    return 0;
+  }
+  return dramaSpilbergMovies.length;
+}
 
 // Iteration 5: Alphabetic Order - Order by title and print the first 20 titles
 
+function orderAlphabetically(arr) {
+  const titlesArr = arr.map(movie => movie.title);
+  return titlesArr.sort().slice(0, 20);
+}
+
 // Iteration 6: Time Format - Turn duration of the movies from hours to minutes
 
+function turnHoursToMinutes(arr) {
+  const minutesMovies = JSON.parse(JSON.stringify(arr));
```

Comment:
Very nice idea here. If you understand the logic behind using these JSON methods, you're good :)